### PR TITLE
Allow mesh adaptivity with interface kernels

### LINKED ISSUE #10977: Allow mesh adaptivity with interface kernels
## Rationale
Mesh adaptivity works with interface kernels as long as stateful material properties aren't used

## Description
Currently MOOSE errors if interface kernels are used with adaptivity, which is an unnecessary restriction

## Impact
With this check removed, users can use adaptivity with interface kernels

### PR DESCRIPTION
**Pull request by lindsayad**
However, we install an error check which errors if the user attempts to use stateful properties, adaptivity, and interface or dg kernels.

Adds two tests that demonstrate use of adaptivity with dg and interface kernels as long as regular material properties are used. Two other tests error if dg and interface kernels are used with adaptivity and stateful properties

Closes #10977

### CODE REVIEW

File: framework/include/base/FEProblemBase.h [hunk 1]
```diff
@@ -1569,6 +1569,9 @@ class FEProblemBase : public SubProblem, public Restartable
   /// At or beyond initialSteup stage
   bool _started_initial_setup;
 
+  /// Whether the problem has dgkernels or interface kernels
+  bool _has_internal_residual_objects;
```
Comment (friedmud) [inline framework/include/base/FEProblemBase.h]: `internal` isn't quite the right word here.  `internal_edge`?

File: test/tests/dgkernels/adaptivity/tests [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[Tests]
```
Comment (friedmud) [inline test/tests/dgkernels/adaptivity/tests]: These two tests need to use the new `requirements` syntax to name the requirement and link back to the issue.

File: test/tests/interfacekernels/adaptivity/tests [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[Tests]
```
Comment (friedmud) [inline test/tests/interfacekernels/adaptivity/tests]: These tests need the "requirements" syntax as well

**Review by friedmud — DISMISSED** [review]
Just some small changes.

**Comment by lindsayad** [comment]
Documentation for this PR will have to wait until the documentation freeze is over

**Review by friedmud — APPROVED** [review]
👍

**Pull request merged**